src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+    of one being in the volume. This can be overwritten after initialization time,
+    but if it is too low a `RejectionException` is likely and if it is too high sampling
+    can be prolonged.
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if needsSampling(self):
+            return
+
+        # Ensure the mesh is watertight so volume is well defined
+        if not self._mesh.is_volume:
+            raise ValueError("A MeshVolumeRegion cannot be defined with a mesh that does not have a well defined volume.")
+
+        # Compute how many samples are necessary to achieve 99% probability
+        # of success when rejection sampling volume.
+        p_volume = self._mesh.volume/self._mesh.bounding_box.volume
+
+        if p_volume > 0.99:
+            self.num_samples = 1
+        else:
+            self.num_samples = min(1e6, max(1, math.ceil(math.log(0.01, 1 - p_volume))))
+
+    # Property testing methods #
+    def intersects(self, other, triedReversed=False):
+        """Check if this region intersects another.
+
+        This function handles intersect calculations for `MeshVolumeRegion` with:
+        * `MeshVolumeRegion`
+        * `MeshSurfaceRegion`
+        * `PolygonalFootprintRegion`
+        """
+        # Check if region is fixed, and if not returns a default implementation
+        if needsSampling(self):
+            return super().intersects(other)
+
+        if isinstance(other, MeshVolumeRegion):
+            # PASS 1
+            # Check if bounding boxes intersect. If not, volumes cannot intersect.
+            # For bounding boxes to intersect there must be overlap of the bounds
+            # in all 3 dimensions.
+            range_overlaps = [(self.mesh.bounds[0,dim] <= other.mesh.bounds[1,dim]) and \
+                              (other.mesh.bounds[0,dim] <= self.mesh.bounds[1,dim]) \
+                              for dim in range(3)]
+            bb_overlap = all(range_overlaps)
+
+            if not bb_overlap:
+                return False
+
+            # PASS 2
+            # Compute inradius and circumradius for a candidate point in each region,
+            # and compute the inradius and circumradius of each point. If the candidate 
+            # points are closer than the sum of the inradius values, they must intersect.
+            # If the candidate points are farther apart than the sum of the circumradius
+            # values, they can't intersect.
+
+            # Get a candidate point from each mesh. If the center of the object is in the mesh use that.
+            # Otherwise try to sample a point as a candidate, skipping this pass if the sample fails.
+            if self.containsPoint(Vector(*self.mesh.bounding_box.center_mass)):
+                s_candidate_point = Vector(*self.mesh.bounding_box.center_mass)
+            elif len(samples:=trimesh.sample.volume_mesh(self.mesh, self.num_samples)) > 0:
```

Comment:
```suggestion
            elif samples := trimesh.sample.volume_mesh(self.mesh, self.num_samples):
```
(In general `if X:` is better than `if len(X) > 0:` for containers. See PEP8 for example.)

Reply:
`ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`

Reply:
Oh, sorry, I didn't realize the return value was a NumPy array. Then I guess we have to have the explicit `len`.